### ISSUE TITLE
NeonVM: v0.8.0 has no manifest for running without multus

### ISSUE THREAD

**Issue by sharnoff** [neondatabase/autoscaling#301]
Ref https://github.com/neondatabase/cloud/pull/5177#discussion_r1209421266

I think this can be fixed by going back to the "neonvm.yaml" / "neonvm-multus.yaml" pair, with the current "neonvm.yaml" going back to "neonvm-multus.yaml".

**Comment by sharnoff** [comment]
This was resolved by https://github.com/neondatabase/autoscaling/pull/340.

**Comment by hlinnaka** [comment]
There's a reference to this issue at https://github.com/neondatabase/cloud/blob/main/local-k3d-cluster/Makefile#L61. Can we update the version in the Makefile now?

**Comment by sharnoff** [comment]
We can't _just_ upgrade unfortunately; some changes required in the way that the k3d cluster is set up (see the k3d directory, particularly k3d/config.yaml)

**Comment by hlinnaka** [comment]
I upgraded them in my massive PR here: https://github.com/neondatabase/cloud/pull/6082. But I had to patch the yaml file to remove the NetworkAttachmentDefinition to make it work:
```
# FIXME: Ok I guess we need Multus after all? This removes the NetworkAttachmentDefinition
# resources that were causing trouble.
objects = read_yaml_stream('local-k3d-cluster/neonvm-{}.yaml'.format(neonvm_version))
new_objects = []
for o in objects:
    if o['kind'] != 'NetworkAttachmentDefinition':
        new_objects.append(o)
k8s_yaml(encode_yaml_stream(new_objects))
k8s_yaml('local-k3d-cluster/autoscale-scheduler-{}.yaml'.format(autoscaling_version))
k8s_yaml('local-k3d-cluster/autoscaler-agent-{}.yaml'.format(autoscaling_version))
```.
However, I don't really know what I'm doing. Looking at `k3d/config.yaml`, I don't understand what that does either. Should we change the k3d creation scripts in the 'cloud' repo to do the same?

**Comment by sharnoff** [comment]
After briefly looking at that PR, I would recommend against copying the autoscaling yaml files into the repo directly. Updating to the next version of autoscaling (something that we _should_ be doing every 1-2 weeks) needs to be as simple as bumping the version number somewhere.

@cicdteam knows more about k3d/config.yaml and why it does what it does.

**Comment by cicdteam** [comment]
`k3d/config.yam`l used to disable built-in CNI (flannel) as Cilium-CNI used in this configuration. Chi was changed become built-in doesn't work with other CNIs (and with Multus-CNI as well) 
Also it disables some features not used in autoscaling (traefik, load balancer and so on)

**Comment by cicdteam** [comment]
btw, `kind` is more native k8s than `k3d` (from a network point of view)

**Comment by hlinnaka** [comment]
> btw, kind is more native k8s than k3d (from a network point of view)

Should we switch to 'kind' in the console repo?

**Comment by hlinnaka** [comment]
So, what should we do here? Sooner or later we need to update the autoscaling version that we use in the console local k8s cluster. Otherwise, it will diverge more and more from the real environment, and we cannot use the local dev environment for testing changes to autoscaling components together with the console and control plane.

Can we do something with kustomize to make the neonvm yaml file work with k3d? Or?

**Comment by hlinnaka** [comment]
Do we use multus in production?

**Comment by sharnoff** [comment]
> Should we switch to 'kind' in the console repo?

Either switch to kind or use a k3d config that allows running with multus. That's ultimately a question for the cloud repo stakeholders.

> Sooner or later we need to update the autoscaling version that we use in the console local k8s cluster

So _currently_ the "autoscaling" vs "NeonVM" versions are separate for this reason, so autoscaling can actually be updated. But yes, we should fix this.

> Can we do something with kustomize to make the neonvm yaml file work with k3d? Or?

Last time I checked, no. IIRC, I tried manually running without multus and the VM controller required it (?). I may be misremembering though.

> Do we use multus in production?

Yes. See also: my [response](https://neondb.slack.com/archives/C03TN5G758R/p1690734589810149?thread_ts=1690480950.803599&cid=C03TN5G758R) to your questions in #vm-autoscaling.
